fix: Add support for cachix using base16 encoding

kalbasit opened the pull request:
### TL;DR

Added support for handling sha256/base-16 encoded FileHash in NAR files

### What changed?

- Added a new test case `Nar5` that includes a FileHash encoded in sha256/base-16 format
- Created a new test file to verify NAR file parsing and validation
- Added `Nar5` to the list of test entries
- Updated go-nix dependency to latest version

### How to test?

Run the new test suite:
```bash
go test ./testdata -v
```

### Why make this change?

Some NAR files from cachix use sha256/base-16 encoding for their FileHash instead of the standard format. This addition ensures proper handling and validation of these differently encoded hashes, improving compatibility with cachix cache entries.

kalbasit commented:
@Mic92 @zimbatm go-nix does not currently support base16 (or base64 for that matters). Is there any work pending getting them implemented?

Example of NAR with base16 (included in the PR): https://nix-community.cachix.org/7k3i624w0rfrvcrbbdrw0zrvasywxmz4.narinfo

Mic92 commented:
Not aware of any work in this direction.

zimbatm commented:
@kalbasit take a look at https://github.com/nix-community/go-nix/pull/120. I started the work on an old branch and switched to something else before rebasing it.

kalbasit commented:
> @kalbasit take a look at [nix-community/go-nix#120](https://github.com/nix-community/go-nix/pull/120). I started the work on an old branch and switched to something else before rebasing it.

@zimbatm I rebased your pr in https://github.com/nix-community/go-nix/pull/127 and I can confirm this PR passes with it. Do you mind taking a look at it soon?

kalbasit commented:
* **#70** <a href="https://app.graphite.dev/github/pr/kalbasit/ncps/70?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/kalbasit/ncps/70?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

zimbatm commented:
Merged upstream :+1:

kalbasit commented:
### Merge activity

* **Dec 31, 2:56 PM EST**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/kalbasit/ncps/70).
